File: Assignment2/src/logic/DfaUnion.scala
package logic
import scala.collection.mutable.ArrayBuffer

class DfaUnion {
  var start: DfaUnionNode = null
  var end: DfaUnionNode = null

  var nodes = new ArrayBuffer[DfaUnionNode]

  def getNode(node1: DfaNode, node2: DfaNode): DfaUnionNode = {
    for (node <- nodes) {
      if (node.node1 == node1 && node.node2 == node2) {
        return node;
      }
    }
    return null;
  }

}

File: Assignment2/src/parser/DfaUnion.java
package parser;

import java.util.ArrayList;

public class DfaUnion {
	public DfaUnionNode start;
	public DfaUnionNode end;
	
	public ArrayList<DfaUnionNode> nodes = new ArrayList<DfaUnionNode>();
	
	public DfaUnionNode getNode(DfaNode node1, DfaNode node2) {
		for(DfaUnionNode node : nodes) {
			if(node.node1 == node1 && node.node2 == node2) {
				return node;
			}
		}
		return null;
	}
}
